src/java.base/share/classes/java/lang/Byte.java

```diff
@@ -574,6 +584,116 @@ public static long toUnsignedLong(byte x) {
         return ((long) x) & 0xffL;
     }
 
+    /**
+     * Based on the unsigned comparison returns the greater of two {@code byte} values.
+     *
+     * @param a the first operand
+     * @param b the second operand
+     * @return the greater of {@code a} and {@code b}
+     * @see java.util.function.BinaryOperator
+     * @since 24
+     */
+    public static byte maxUnsigned(byte a, byte b) {
+        return compareUnsigned(a, b) > 0 ? a : b;
+    }
+
+    /**
+     * Based on the unsigned comparison returns the smaller of two {@code byte} values.
+     *
+     * @param a the first operand
+     * @param b the second operand
+     * @return the smaller of {@code a} and {@code b}
+     * @see java.util.function.BinaryOperator
+     * @since 24
+     */
+    public static byte minUnsigned(byte a, byte b) {
+        return compareUnsigned(a, b) < 0 ? a : b;
+    }
+
+    /**
+     * Saturating addition of two {@code byte} values,
+     * which returns a {@code Byte.MIN_VALUE} in underflowing or
+     * {@code Byte.MAX_VALUE} in overflowing scenario.
+     *
+     * @param a the first operand
+     * @param b the second operand
+     * @return the sum of {@code a} and {@code b} iff within {@code byte} value range else delimiting {@code Byte.MIN_VALUE/MAX_VALUE} value.
+     * @see java.util.function.BinaryOperator
+     * @since 24
+     */
+    public static byte addSaturating(byte a, byte b) {
+        int res = a + b;
+        if (res > Byte.MAX_VALUE) {
+            return Byte.MAX_VALUE;
+        } else if (res < Byte.MIN_VALUE) {
+            return Byte.MIN_VALUE;
+        } else {
+           return (byte)res;
+        }
+    }
+
+    /**
+     * Saturating subtraction of two {@code byte} values,
+     * which returns a {@code Byte.MIN_VALUE} in underflowing or
+     * {@code Byte.MAX_VALUE} in overflowing scenario.
+     *
+     * @param a the first operand
+     * @param b the second operand
+     * @return the difference between {@code a} and {@code b} iff within {@code byte} value range else delimiting {@code Byte.MIN_VALUE/MAX_VALUE} value.
+     * @see java.util.function.BinaryOperator
+     * @since 24
+     */
+    public static byte subSaturating(byte a, byte b) {
+        byte res = (byte)(a - b);
```

Comment:
Yes, core libs also have {add/subtract}Exact API which instead of saturating over / underflowing values throws ArithmeticException. Streamlining overflow checks for saturating long operations on the same lines to address Joe's concerns on new constants.